simulator/risc_v/t/unit_test.cpp

```diff
@@ -155,3 +155,45 @@ TEST_CASE( "RISCV lq/sq")
     memory->load_store( &load);
     CHECK( narrow_cast<uint64>( load.get_v_dst()) == 0xf);
 }
+
+template<typename T>
+struct TestData {
+    T src1, src2,dst;
+};
+
+TEST_CASE( "RISCV slo32")
+{
+    CHECK( RISCVInstr<uint32>( 0x20E797B3).get_disasm() == "slo $a5, $a5, $a4");
+    RISCVInstr<uint32> instr( "slo", 0);
```

Comment:
Please move ctor into the loop. There is no warranty of correct result if 'execute()' is called twice, and some day it may produce errors, and we don't want to spend time to debug them.